src/stream/mod.rs

```diff
@@ -1091,16 +1091,21 @@ where
 }
 
 #[cfg(feature = "std")]
-pub struct ReadStream<R> {
+pub struct ReadStream<R, P> {
     bytes: Bytes<R>,
+    phantom: PhantomData<P>
 }
 
 #[cfg(feature = "std")]
-impl<R: Read> StreamOnce for ReadStream<R> {
+impl<R, P> StreamOnce for ReadStream<R, P>
+where
+    R: Read,
+    P: Ord + PartialEq + Clone
+{
     type Token = u8;
     type Range = &'static [u8];
-    type Position = usize;
-    type Error = Errors<Self::Token, Self::Range, usize>;
+    type Position = P;
+    type Error = Errors<Self::Token, Self::Range, P>;
```

Comment:
Thinking about it and the actual problem is that `ReadStream` uses `easy::Errors` instead of it's own, dedicated type. I went ahead and did that change + a refactor in https://github.com/Marwes/combine/pull/273 .

You should be able to use that + wrap your stream with `easy::Stream` to get the same behaviour as with this PR.